auth.go

```diff
@@ -19,3 +19,16 @@ func (c *AuthConfig) encode() string {
 	json.NewEncoder(&buf).Encode(c)
 	return base64.URLEncoding.EncodeToString(buf.Bytes())
 }
+
+// ConfigFile holds parameters for authenticating during a BuildImage request
+type ConfigFile struct {
+	Configs  map[string]AuthConfig `json:"configs,omitempty"`
+	rootPath string
+}
+
+// encode the configuration struct into base64 for the X-Registry-Config header
+func (c *ConfigFile) encode() string {
+	var buf bytes.Buffer
+	json.NewEncoder(&buf).Encode(c)
```

Comment:
I think it's best to check for the error here, and make this method return `(string, error)`